### PR TITLE
ARROW-7537: [CI][R] Nightly macOS autobrew job should be more verbose if it fails

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

**Comment by nealrichardson** [comment]
@ursabot crossbow submit macos-r-autobrew

File: r/tools/autobrew [hunk 1]
```diff
@@ -62,3 +62,5 @@ unset HOMEBREW_NO_AUTO_UPDATE
 
 # Hack to work around segfault on CRAN server
 # rm -f tests/testthat.R
+
+exit 1
```
Comment (kszucs) [inline r/tools/autobrew]: The log is correctly displayed.

Comment (nealrichardson) [reply r/tools/autobrew]: Yes, I didn’t trigger a failure in the right place, will try again

**Comment by nealrichardson** [comment]
@ursabot crossbow submit macos-r-autobrew

**Comment by nealrichardson** [comment]
That crossbow run confirmed the expected behavior, so I've reverted the intentional error to trigger it, and I've copied in some env vars we use in the other R test suites. Merging.

**Pull request closed without merging**